includes/class-wc-cache-helper.php

```diff
@@ -42,14 +42,34 @@ public static function init() {
 	 */
 	public static function additional_nocache_headers( $headers ) {
 		$agent = isset( $_SERVER['HTTP_USER_AGENT'] ) ? wp_unslash( $_SERVER['HTTP_USER_AGENT'] ) : ''; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+
+		$set_cache = false;
+
 		/**
 		 * Allow plugins to enable nocache headers. Enabled for Google weblight.
 		 *
-		 * @see   https://support.google.com/webmasters/answer/1061943?hl=en
 		 * @param bool $enable_nocache_headers Flag indicating whether to add nocache headers. Default: false.
 		 */
-		if ( false !== strpos( $agent, 'googleweblight' ) || apply_filters( 'woocommerce_enable_nocache_headers', false ) ) {
+		if ( apply_filters( 'woocommerce_enable_nocache_headers', false ) ) {
```

Comment:
I'm kind of curious why the code (before your changes) wasn't framed in this sort of way:

```php
if ( false !== strpos( $agent, 'googleweblight' ) ) {
	$set_cache = true;
}

/**
 * Allow plugins to enable nocache headers. Enabled for Google weblight.
 *
 * @param bool $enable_nocache_headers Flag indicating whether to add nocache headers. Default: false.
 */
$set_cache = (bool) apply_filters( 'woocommerce_enable_nocache_headers', $set_cache );
```

If only because the way the filter's docblock reads to me is that `$set_cache` will *already* be true if the request was made by Google Weblight...but really what's happening in the [current code](https://github.com/woocommerce/woocommerce/blob/5.3.0/includes/class-wc-cache-helper.php#L51) is, we force it to true if the request was made by Google Weblight—regardless of what comes through the filter (and I'm just noting this to give context for my next few points).

> We could also alternatively introduce a filter to disable the cache if they so choose.

Couldn't we do this now? If for back compat reasons we don't want to change the way the logic works for Weblight, we could still do something like I outlined above but focused on Chrome and the cart page. That way it would be easy for merchants to unwind this change if necessary, at no extra expense in terms of adding or triggering new hooks.

I'm also wondering if we should update the docblock, because we're adding this new exception for Chrome but we're not documenting it there (whereas it's documented for Weblight).

Reply:
Yeah not sure why it was like that from before but I just reorganized it to a way which was more readable to me. Before it was kind of confusing not to mention the added conditional for Chrome and cart.